text.go

```diff
@@ -528,25 +528,18 @@ func (t *Text) MostCommonFontFace() FontFace {
 	return family.Face(size*ptPerMm, col, style, variant)
 }
 
-// ToPaths makes a path out of the text, with x,y the top-left point of the rectangle that fits the text (ie. y is not the text base)
-func (t *Text) ToPaths() ([]*Path, []color.RGBA) {
-	paths := []*Path{}
-	colors := []color.RGBA{}
+// RenderAsPath renders the text (and its decorations) converted to paths (calling r.RenderPath)
+func (t *Text) RenderAsPath(r Renderer, m Matrix) {
+	style := DefaultStyle
 	for _, line := range t.lines {
 		for _, span := range line.spans {
 			p, _, col := span.ToPath(span.width)
 			p = p.Translate(span.dx, line.y)
-			paths = append(paths, p)
-			colors = append(colors, col)
-		}
-		for _, deco := range line.decos {
-			p := deco.face.Decorate(deco.x1 - deco.x0)
-			p = p.Translate(deco.x0, line.y)
-			paths = append(paths, p)
-			colors = append(colors, deco.face.Color)
+			style.FillColor = col
+			r.RenderPath(p, style, m)
 		}
 	}
-	return paths, colors
+	t.RenderDecoration(r, m)
```

Comment:
I think this would have the same problem as before, in the second line overlaps the first line and both have an underline, then the underline of the first line will overlap the second line. The only way is the draw decorations after every text line immediatly, and not as a batch after all the text lines.

Reply:
I attempted something with `text.WalkLines`. I am not 100% convinced yet: feel free to play with it and suggest any improvemt :smile: